OpenRA.Mods.Common/Projectiles/SoundBeam.cs

```diff
@@ -0,0 +1,181 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Projectiles
+{
+	[Desc("Beam projectile that travels in a straight line.")]
+	public class SoundBeamInfo : IProjectileInfo
+	{
+		[Desc("Projectile speed in WDist / tick, two values indicate a randomly picked velocity per beam.")]
+		public readonly WDist[] Speed = { new(128) };
+
+		[Desc("The number of ticks between the beam causing warhead impacts in its area of effect.")]
+		public readonly int DamageInterval = 1;
+
+		[Desc("Equivalent to sequence ZOffset. Controls Z sorting.")]
+		public readonly int ZOffset = 0;
+
+		[Desc("The minimum distance the beam travels.")]
+		public readonly WDist MinDistance = WDist.Zero;
+
+		[Desc("Visual Width of the beam")]
+		public readonly WDist Width = new(650);
+
+		[Desc("Visual lenght in ticks")]
+		public readonly int Lenght = 4;
+
+		[Desc("Damage modifier applied at each range step.")]
+		public readonly int[] Falloff = { 100, 100 };
+
+		[Desc("Ranges at which each Falloff step is defined.")]
+		public readonly WDist[] Range = { WDist.Zero, new(int.MaxValue) };
+
+		[Desc("The maximum/constant/incremental inaccuracy used in conjunction with the InaccuracyType property.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Controls the way inaccuracy is calculated. Possible values are" +
+		"'Maximum' - scale from 0 to max with range," +
+		"'PerCellIncrement' - scale from 0 with range" +
+		"'Absolute' - use set value regardless of range.")]
+		public readonly InaccuracyType InaccuracyType = InaccuracyType.Maximum;
+
+		[Desc("Can this projectile be blocked when hitting actors with an nameof(BlocksProjectiles) trait.")]
+		public readonly bool Blockable = false;
+
+		[Desc("Does the beam follow the target.")]
+		public readonly bool TrackTarget = false;
+
+		[Desc("Color of the beam.")]
+		public readonly Color Color = Color.SkyBlue;
+
+		public IProjectile Create(ProjectileArgs args)
+		{
+			return new SoundBeam(this, args, Color);
+		}
+	}
+
+	public class SoundBeam : IProjectile, ISync
```

Comment:
The OG effect (which i'm tempted to implement after this PR) isn't a beam, so could we rename this to something like "SonicBlast"?

Reply:
We should also move this into the Mods.d2k assembly since this is going to be quite d2k-specific once complete.

Reply:
I'm not sure that's necessary

Reply:
I have a commit ready to go now adding the OG zoom effect. It is simple and fast, but not very flexible (you can set the global size and zoom of the effect, but can't set different values for different weapons or add tints etc). Adding extra flexibility for modders would impact both the simplicity and perf, which I don't think can be justified.

https://github.com/user-attachments/assets/3caddc30-1946-41b6-b070-93fea58ab73c